packages/core/src/Renderer.ts

```diff
@@ -127,7 +128,7 @@ export class Renderer {
     _renderTextNode(context: RenderingContext): RenderingContext {
         // If the node has a format, render the format nodes first.
         const renderedFormats = [];
-        const firstChar = context.currentVNode;
+        const firstChar = context.currentVNode as CharNode;
```

Comment:
How can we be sure that the currentVNode is a CharNode?
We might need a type guard to check if it is really one, otherwise, a bug might slip here.

Reply:
We are sure because this method is called after checking that the node we are rendering is a text node.